contrib/dracut/90zfs/zfs-generator.sh.in

```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-GENERATOR_DIR="$1"
+GENERATOR_DIR="$2"
```

Comment:
I see, so we are flat out overrriding the existing sysroot.mount here.

This will fail.  There is no Where= in the generator output.

I think it is better to just use a conf.d directory.   Alternatively, write a complete generator unit like the dmsquash module does.